gsoc-2024/work-product/Maria-Pana-Multiboot2-Support-in-Unikraft/Maria-Pana-Multiboot2.md

```diff
@@ -0,0 +1,148 @@
+# Multiboot2 support in Unikraft
+
+<img width="100px" src="https://summerofcode.withgoogle.com/assets/media/gsoc-generic-badge.svg" align="right" />
+
+## Project Overview
+
+When booting an x86 system, the CPU starts in 16-bit mode and the memory is limited to 1MB (in Real Mode).
+The CPU starts executing at a specific memory address known as a reset vector situated at 0xFFFF0 (for Intel 8086 processors) or 0xFFFFFFF0 (for Intel 80386 and later).
+The initial instructions found here are part of the BIOS, stored in flash memory on the motherboard.
+The BIOS's first goal is to run self test and initialization routines on hardware.
+During this phase, the BIOS sets up standardized interfaces (e.g. legacy BIOS interrupts or UEFI routines) that software that succeeds it (bootloader or operating system) will use to understand the computer's resources without needing special drivers for each different motherboard model.
+
+However, having the kernel directly deal with the interfaces installed by the BIOS unnecessarily increases complexity and maintenance effort.
+This is where the bootloader comes in.
+After its initial routines, BIOS looks for bootable devices, such as a Hard Disk.
+If it finds one that has a valid (P)MBR or GPT partitioning scheme with a boot partition, the control is passed to the bootloader.
+This will then load the OS kernel into memory either by using the interfaces offered by the BIOS or its own drivers, and transfer control to it.
+
+There are multiple boot protocols, some already supported by Unikraft.
```

Comment:
This is the first time you discuss the term "boot protocol". Tell what a boot protocol is, beforehand.